persistence/persistence-api/src/main/java/org/cbioportal/persistence/util/CustomEhCachingProvider.java

```diff
@@ -227,13 +227,15 @@ public void detectCacheConfigurationErrorsAndLog() {
                 File persistenceDirectory = new File(persistencePath);
                 boolean accessible = false;
                 try {
-                    if (persistenceDirectory.isDirectory()) {
+                    if (persistenceDirectory.isDirectory() && persistenceDirectory.canWrite()) {
```

Comment:
I added this check to prevent silently failing when a valid persistence directory is provided but the user does not have write permissions to this directory